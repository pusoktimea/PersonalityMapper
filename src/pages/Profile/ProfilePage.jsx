import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import Loader from 'components/Loader';
import cx from 'classnames';

import ProfileForm from './ProfileForm';

import './profile-page.scss';

class ProfilePage extends PureComponent {
  static propTypes = {
    onGetProfile: PropTypes.func.isRequired,
    isLoading: PropTypes.bool.isRequired,
    profile: PropTypes.object.isRequired,
    onUpdateProfile: PropTypes.func.isRequired,
    isUpdateInProgress: PropTypes.bool.isRequired,
    isProfileLoaded: PropTypes.bool.isRequired,
    isSideBarMinimised: PropTypes.bool
  }

  componentWillMount() {
    if (!this.props.isProfileLoaded) {
      this.props.onGetProfile();
    }
  }

  render() {
    const {
      isLoading,
      profile,
      onUpdateProfile,
      isUpdateInProgress,
      isSideBarMinimised
    } = this.props;
    const baseClass = 'main-content';

    return (
      <div className={cx('profile-page', baseClass, isSideBarMinimised && `${baseClass}--stretched`)}>
        <h2 className="title">Profile</h2>
        {
          isLoading ?
            <Loader /> :
            <ProfileForm
              profile={profile}
              onUpdateProfile={onUpdateProfile}
              isUpdateInProgress={isUpdateInProgress}
            />
        }
      </div>
    );
  }
}

export default ProfilePage;
