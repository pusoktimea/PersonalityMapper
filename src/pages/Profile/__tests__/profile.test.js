import React from 'react';
import renderer from 'react-test-renderer';

import ProfilePage from '../ProfilePage';

const pageProps = {
  onGetProfile: jest.fn(),
  isLoading: false,
  profile: {
    email: 'email@test.com',
    full_name: 'John Doe',
    phone: '123-555-1234',
    user_name: 'john_doe',
    uid: '1'
  },
  onUpdateProfile: jest.fn(),
  isUpdateInProgress: false,
  isProfileLoaded: true,
  isSideBarMinimised: true
};

describe('Profile Page test', () => {
  it('renders and matches snapshot', () => {
    expect(
      renderer.create(
        <ProfilePage {...pageProps} />
      ).toJSON()
    ).toMatchSnapshot();
  });
});
