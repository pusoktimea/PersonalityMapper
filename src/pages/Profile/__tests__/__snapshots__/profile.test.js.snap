// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page test renders and matches snapshot 1`] = `
<div
  className="profile-page main-content main-content--stretched"
>
  <h2
    className="title"
  >
    Profile
  </h2>
  <div
    className="activate-panel profile-page_form"
  >
    <div
      className="activate-panel_body profile-page_form"
    >
      <h3
        className="profile-page_form_title"
      >
        General Information
      </h3>
      <label
        className="activate-label"
      >
        <span>
          Full Name
        </span>
        <div
          className="activate-input activate-input--dark"
        >
          <input
            className="activate-input_input"
            disabled={false}
            name="full_name"
            onChange={[Function]}
            type="text"
            value="John Doe"
          />
        </div>
      </label>
      <label
        className="activate-label"
      >
        <span>
          Phone
        </span>
        <div
          className="activate-input activate-input--dark"
        >
          <input
            className="activate-input_input"
            disabled={false}
            name="phone"
            onChange={[Function]}
            type="text"
            value="123-555-1234"
          />
        </div>
      </label>
      <label
        className="activate-label"
      >
        <span>
          Email
        </span>
        <div
          className="activate-input activate-input--dark"
        >
          <input
            className="activate-input_input"
            disabled={false}
            name="email"
            onChange={[Function]}
            type="email"
            value="email@test.com"
          />
        </div>
      </label>
      <h3
        className="profile-page_form_title"
      >
        Change Password
      </h3>
      <label
        className="activate-label"
      >
        <span>
          New Password
        </span>
        <div
          className="activate-input activate-input--dark"
        >
          <input
            className="activate-input_input"
            disabled={false}
            name="password"
            onChange={[Function]}
            placeholder="Enter New Password"
            type="password"
            value=""
          />
        </div>
      </label>
      <label
        className="activate-label"
      >
        <span>
          Confirm Password
        </span>
        <div
          className="activate-input activate-input--dark"
        >
          <input
            className="activate-input_input"
            disabled={false}
            name="confirm_password"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
            value=""
          />
        </div>
      </label>
      <button
        className="activate-button activate-button--primary activate-button--size-default profile-page_form_button"
        disabled={false}
        onClick={[Function]}
        to={undefined}
      >
        <span
          className="activate-icon fa fa-check-square-o"
          id={undefined}
          onClick={undefined}
        />
        Update Profile
      </button>
    </div>
  </div>
</div>
`;
